### PR TITLE
Multi address oracle

### PR DESCRIPTION
**Pull request by thodges-gh**
Finishes [#162199318](https://www.pivotaltracker.com/story/show/162199318).

### CODE REVIEW

File: examples/ropsten/contracts/RopstenConsumer.sol [hunk 1]
```diff
@@ -248,6 +252,22 @@ library ChainlinkLib {
       self.buf.buf);
   }
 
+  function encodeForCoordinator(
+    Run memory self,
+    uint256 _clArgsVersion
+  ) internal pure returns (bytes memory) {
+    return abi.encodeWithSelector(
+      coordinatorRequestDataFid, // need to change
```
[the remark this answers is not in the document]
Comment (se3000) [reply examples/ropsten/contracts/RopstenConsumer.sol]: I think it's out of date. I removed it in a separate PR.

File: examples/ropsten/contracts/Oracle.sol [hunk 2]
```diff
@@ -116,17 +116,45 @@ library SafeMath {
   }
 }
 
+// File: contracts/interfaces/OracleInterface.sol
```
[the remark this answers is not in the document]
Comment (se3000) [reply examples/ropsten/contracts/Oracle.sol]: I believe it's the output of the solidity flattener, which we use to make contracts easy to import in Remix.

File: examples/ropsten/contracts/Oracle.sol [hunk 3]
```diff
@@ -262,6 +300,11 @@ contract Oracle is Ownable {
     _;
   }
 
+  modifier onlyAuthorizedNode() {
+    require(authorizedNodes[msg.sender] == true || msg.sender == owner, "Not an authorized node to fulfill requests");
```
[the remark this answers is not in the document]
Comment (thodges-gh) [reply examples/ropsten/contracts/Oracle.sol]: @j16r @se3000 I have this change ready, but consider this type of usage: the owner adds their nodes' addresses to the `authorizedNodes` mapping. The owner's address is a normal wallet, or even a smart contract, but not a node itself. The `onlyAuthorizedNode` modifier is only used on the `fulfillData` method, which can only be called by the node. Given this usage, if we short-circuit on owner first, then it's more expensive for each node to call `fulfillData`.

[the remark this answers is not in the document]
Comment (j16r) [reply examples/ropsten/contracts/Oracle.sol]: Cheaper for the most common case, makes sense 👍

**Pull request merged**